### DISCUSSION TITLE
This video errors out when I try to download it

### DISCUSSION THREAD

**Discussion by whocrazy** [aclap-dev/video-downloadhelper#31] [Help and Q&A] [answered]
### Your question

SideDownload error: 
[aac_adtstoasc @ 0000025c304074c0] Error parsing ADTS frame header!
[mp4 @ 0000025c30dbdd00] Error applying bitstream filters to an output packet for stream #1: Invalid data found when processing input
av_interleaved_write_frame(): Invalid data found when processing input
[out#0/mp4 @ 0000025c3040a240] Error muxing a packet
Due to the emotive subject matter, I am not sure if it is OK to put the URL to the video on here, some people might find the interview hard to watch.  Oh, it's not anything xxx, or rude or anything like that
Is there some person I can message in private the URL of the video and they can try and find out why I am getting this error?
This error doesn't happen with all the videos, just some of them.

### Addon & CoApp version and details

version: 8.2.0.18a1
channel: beta
build date: 2024-02-01
build options: linuxlic: false, noyt: false, browser: firefox
lang: en-US
license: accepted
key (only 16 first characters): 64a87ee44efb4545
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:122.0) Gecko/20100101 Firefox/122.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.10",
  "binary": "C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "C:\\Users\\MySystem",
  "converterBinary": "C:\\Program Files\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "6.0"
}

**Comment by jcfrog** [comment]
hi @whocrazy ,
you can send the link at support at downloadhelper dot net

**Reply by whocrazy** [reply]
Done, thank you.  Hopefully  there will be a way to download these videos.

**Comment by paulrouget** [comment]
Also please try a more recent coapp: https://github.com/aclap-dev/video-downloadhelper/wiki/CoApp-Installation

**Reply by whocrazy** [reply]
version: 8.2.0.18a1
channel: beta
build date: 2024-02-01
build options: linuxlic: false, noyt: false, browser: firefox
lang: en-US
license: accepted
key (only 16 first characters): 64a87ee44efb4545
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:122.0) Gecko/20100101 Firefox/122.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.14",
  "binary": "C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "C:\\Users\\MySystem",
  "converterBinary": "C:\\Program Files\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "63b936e978"
}

**Reply by whocrazy** [reply]
btw, the way I currently get around the issue is to download the m3u8 manually and then use a mass downloader extension like down them all to grab the segments, then merge them like this
First, I've got a batch file that does this.
@echo off
(for %%i in (*.ts) do @echo file '%%i') > input.txt
Then, the command line goes like this:
ffmpeg -f concat -safe 0 -i input.txt -metadata service_name=" " -metadata service_provider=" " -c copy output.ts
OK, it's messy, I know, but it's the only way I can download these videos.
If there is no fix to be found, could you possibly look at implementing this feature into VDH?
Thanks.

**Comment by GrampaWildWilly** [comment]
I would like to test this URL.  If you say viewer discretion is advised, then there should be no harm in posting the URL here in public.  I usually don't actually watch the videos I download when I'm investigating a problem so your concern for my sensitivities is appreciated but not necessary.  Understand that I don't work for VDH.  I'm just another user.  I was going to say I'm just another user like you but . . . well, you've just revealed elsewhere something that makes you extraordinary.  So I'd just like to have that URL so I can see what happens & perhaps suggest some solutions.

**Comment by whocrazy** [comment]
OK, but, again, for others on here, I warn, viewer discretion is advised.
https://live.childrenshealthdefense.org/chd-tv/events/the-chd-bus-tour/my-30-year-daughter-was-killed-by-a-hospital/

**Reply by GrampaWildWilly** [reply]
I visited that page & the VDH menu was empty.  I clicked the play button on the video & that didn't actually launch playback, which is a bit odd.  But that seemed to be enough to make VDH show something in its menu.  There were just 3 mp4 variants of various resolutions.  I downloaded the 1920x1080 variant without incident.  The download took 1:17, quite fast.  The resulting file is an mp4 of size 1.30G, duration 30:19, resolution 1920x1080, unremarkable but acceptable video & audio properties.  VLC played it just fine.  I skimmed it & there was both audio & video at each of my sample points.

My configuration & yours are the same.  I can't explain what's happening to you.  I have occasionally encountered a muxing error while listening to a radio broadcast online.  It's annoying, but simply restarting the stream clears the error . . . until it happens again.  It appears to be a glitch at that radio station.  I don't encounter the problem elsewhere.

Do you encounter this problem on every attempt to download this one video?  Are you finding that it is a repeatable error?  Are you getting the same error on other content from this web site?

One other possibility is your antivirus & firewall could be interfering.  I hesitate to suggest this because I believe you are not generally having problems downloading, are you?  But others have reported mysterious errors that are cured by making exceptions for the CoApp in the antivirus & the firewall.  That's discussed over here:

https://github.com/aclap-dev/video-downloadhelper/discussions/14

Failing that, I think perhaps somebody like @paulrouget might need to intervene here.

**Reply by paulrouget** [reply, marked as answer]
I'm getting the same error as @whocrazy but only when downloading the low resolution version. I'm pretty sure the video source for low resolution is corrupted.

@whocrazy do you mind testing with the high resolution variant and confirm that it's working for you?

**Reply by GrampaWildWilly** [reply]
Oh yeah.  I just got the error with the third variant on the VDH menu, too.  That's the lowest resolution on offer.  The first one, the 1920x1080 variant, worked fine for me a little while ago.  Our blind pal here wouldn't care about resolution.  He's only listening to the content.  That's interesting.  Which puts me in mind of the new feature you can find on the secondary VDH menu for Audio-only download.  That might be something @whocrazy would find more appropriate for most content.  Am I right?

But the download of the audio did something really weird here.  It got to 8% and stopped dead.  Resource Monitor no longer showed ffmpeg as an active task.  Task Manager did show a Firefox thread for ffmpeg but it was using 0% CPU, 0% Disk, & 0% Network.  It was just using some amount of memory, I forget how much, something small like 5M.  As I was going back & forth & looking at things, deciding what I was going to do next, oops, up pops the VDH completion message.  So for reasons I can't explain, the download completely stopped for about 2 minutes, then resumed & completed without error.  I did get an mp3 of just the sound track of this video.

I have to say I don't know if this is a problem with VDH or with the web site.  I don't generally want to download only the audio track of something.  I suspect @whocrazy might start using it.  Maybe you can collect some experience with that & let us know if this inordinately long delay is something that happens on various different sites.  That would help identify whether it's just this video & this site that are weird or it's a more general problem.

**Reply by GrampaWildWilly** [reply]
My test above was of the sound track of the 1920x1080 variant.  I decided to try the sound track of the 852x480 variant.  It did the same thing.  It stopped at 7% & just sat there for at least 2 minutes.  Then Resource Monitor suddenly showed it resuming at around 15 million bytes per second.  Then it completed without further incident.  This could still just be a quirk of this web site, or just this video on this web site.  We need more data points to draw a conclusion.

**Comment by whocrazy** [comment]
Yes, the high resolution seems to work well.
Just out of curiosity When I use the low resolution setting for this video using ffmpeg, I get a whole slue of errors, including PES packet size mismatch, 
, and this strange ssl error

[tls @ 076d19c0] mbedtls_ssl_read reported connection reset by peer
[https @ 07710ec0] Stream ends prematurely at 1345892, should be 1351908
This always happens at file 00000027.ts
[https @ 07704180] Opening 'https://play.boxcast.com/2024/02/01/181843-sprw2lpygigqmd91vk7e/240p/00000028.ts?Policy=eyJTdGF0ZW1lbnQiOlt7IlJlc291cmNlIjoiaHR0cHM6Ly9wbGF5LmJveGNhc3QuY29tLzIwMjQvMDIvMDEvMTgxODQzLXNwcncybHB5Z2lncW1kOTF2azdlLzI0MHAvKiIsIkNvbmRpdGlvbiI6eyJEYXRlTGVzc1RoYW4iOnsiQVdTOkVwb2NoVGltZSI6MTcwNzI2NDAwMH19fV19&Signature=I442Y30Wn8v7d9zWxHFi72dpOV1BzPvUDv~nZWx7lNzYXvzJlCiVzKzxSQzSp1i7-2hRejTMsu02~5OfSSlEmg-6fEs2XmuC56Sd8Wk~ckRqWNQAKhyaH4-1NK1Xb8RIvEIRaQXDGX28vnDvWBymjykfFDqEmmlMXXtTlK~mhVo3s-zN5qYLZFHGuGOHb3goHoZadLLR~JSxRELWfzChe7RQN~BUAZn9lSMl7AAcQWRXV9mA9GE7x4k2tuspHjyCu7x58H8FVHUAnh94n8AFUrqEKZAxGPcslZzJuvezhGm31W9AK-AvCtajTeYNrg~bi~6sBlKLr43POeSGt2akWQ__&Key-Pair-Id=APKAJ7GUCBQUK6NTWZCA' for reading
[tcp @ 07706b00] ffurl_read returned 0xdfb9b0bb
[hls @ 076b4bc0] keepalive request failed for 'https://play.boxcast.com/2024/02/01/181843-sprw2lpygigqmd91vk7e/240p/00000028.ts?Policy=eyJTdGF0ZW1lbnQiOlt7IlJlc291cmNlIjoiaHR0cHM6Ly9wbGF5LmJveGNhc3QuY29tLzIwMjQvMDIvMDEvMTgxODQzLXNwcncybHB5Z2lncW1kOTF2azdlLzI0MHAvKiIsIkNvbmRpdGlvbiI6eyJEYXRlTGVzc1RoYW4iOnsiQVdTOkVwb2NoVGltZSI6MTcwNzI2NDAwMH19fV19&Signature=I442Y30Wn8v7d9zWxHFi72dpOV1BzPvUDv~nZWx7lNzYXvzJlCiVzKzxSQzSp1i7-2hRejTMsu02~5OfSSlEmg-6fEs2XmuC56Sd8Wk~ckRqWNQAKhyaH4-1NK1Xb8RIvEIRaQXDGX28vnDvWBymjykfFDqEmmlMXXtTlK~mhVo3s-zN5qYLZFHGuGOHb3goHoZadLLR~JSxRELWfzChe7RQN~BUAZn9lSMl7AAcQWRXV9mA9GE7x4k2tuspHjyCu7x58H8FVHUAnh94n8AFUrqEKZAxGPcslZzJuvezhGm31W9AK-AvCtajTeYNrg~bi~6sBlKLr43POeSGt2akWQ__&Key-Pair-Id=APKAJ7GUCBQUK6NTWZCA' with error: 'I/O error' when opening url, retrying with new connection

Can anyone tell me Why this is happening? how come this TLS error doesn't happen if I download the TS file manually using my browser? is it an issue with FFmpeg itself?

**Reply by paulrouget** [reply]
Is that from the coapp ffmpeg?

**Reply by whocrazy** [reply]
No it's not, it's the reino build.
I imagine the coapp won't be any different.

**Comment by whocrazy** [comment]
https://live.childrenshealthdefense.org/chd-tv/events/the-chd-bus-tour/flu-shot-injured-me/
all 3 resolutions give me errors when I try to download them using VDH.
I tried using the coapp version of ffmpeg, and issued this command to download the video
ffmpeg -user_agent "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:122.0) Gecko/20100101 Firefox/122.0" -referer "https://live.childrenshealthdefense.org/chd-tv/events/the-chd-bus-tour/flu-shot-injured-me/" -i "https://play.boxcast.com/p/xttjd6efxfdfex82hfxx/r/4.44081s/580.516s/v/240p-byteranges.m3u8?Expires=1707264000&Signature=uoSYpt9tW1O6XW8rR5NUoM0IQ3631XgCp7IDDX2GXf2gx5lz317~QPTl7m2udbXCyQVo~34gfEkr-H3MPTgOs7ZmP-clyaUJhxc0dUvYYQPr5rua32w3jkUJhV-vbsd8w~8UKy4u0Y6X9LDLHAMQHLrG6ZmZuR~chwXQL-YnzYuRK477FxBF2Z6nQ5qOKPEMTiradzYLwUcdkAArvFRYy8yOnXFQOZOwfg9Fyjk03EJ4EEPD6dJsLpdD0n0JpXFEkg8KVmoIwcbZKGo0x4tfphihKQuEoEk8jElazy7qrGr3HPGKJJfBwEekWjzLNKdcZAfPkVVwJnbLJGaRqSRbfg__&Key-Pair-Id=APKAJ7GUCBQUK6NTWZCA" -c copy "240p-byteranges.ts"
about 5 minutes into the video, when she talks about having chills, the audio, and, I'm guessing the video, suddenly jumps forward about a second.  and then it does it again at 6 minutes and 30 seconds when she mentions upper respiratory issues
Here is the command I use
ffmpeg -user_agent "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:122.0) Gecko/20100101 Firefox/122.0" -referer "https://live.childrenshealthdefense.org/chd-tv/events/the-chd-bus-tour/flu-shot-injured-me/" -i "https://play.boxcast.com/p/xttjd6efxfdfex82hfxx/r/4.44081s/580.516s/v/240p-byteranges.m3u8?Expires=1707264000&Signature=uoSYpt9tW1O6XW8rR5NUoM0IQ3631XgCp7IDDX2GXf2gx5lz317~QPTl7m2udbXCyQVo~34gfEkr-H3MPTgOs7ZmP-clyaUJhxc0dUvYYQPr5rua32w3jkUJhV-vbsd8w~8UKy4u0Y6X9LDLHAMQHLrG6ZmZuR~chwXQL-YnzYuRK477FxBF2Z6nQ5qOKPEMTiradzYLwUcdkAArvFRYy8yOnXFQOZOwfg9Fyjk03EJ4EEPD6dJsLpdD0n0JpXFEkg8KVmoIwcbZKGo0x4tfphihKQuEoEk8jElazy7qrGr3HPGKJJfBwEekWjzLNKdcZAfPkVVwJnbLJGaRqSRbfg__&Key-Pair-Id=APKAJ7GUCBQUK6NTWZCA" -c copy "240p-byteranges.ts"

**Reply by GrampaWildWilly** [reply]
My first bit of advice is don't use the ffmpeg packaged with the CoApp.  It includes modifications that are tailored specifically for use with VDH.  @paulrouget could detail what those are.  I've never looked at the source code of the CoApp, even though it's available.  I doubt it's in a programming language I ever learned.  I tried to use the captive ffmpeg a long time ago & I found it didn't work properly for what I wanted to do.  I would recommend you go to ffmpeg.org & get a regular release of ffmpeg.  As you will see (I never really paid any attention to how often that word pops into common usage) in the logs I'm attaching, I got my build a couple of months ago, in late November.  At ffmpeg.org you will find an array of choices for packages to download.  The one I got was a full quasi-nightly build.  It is my impression that the guy who makes these builds does so every 2 or 3 nights or so.  So it's not quite a nightly build, just almost.  And I always get one of the full packages because I don't know enough to know whether any of the other builds would have everything I might ever want to use it for.  So I get the biggest one I can find.  Better to have more than you need & not use it all than to discover you need more than what you have.

I think you are passing more parameters to ffmpeg than you really need.  I don't know that they are causing you the problems you are seeing.  I would recommend you try again using a full ffmpeg & see if the problems persist.  Either way, I still say you have some parameters you don't need.  The logs I'm attaching show the exact ffmpeg commands I invoked so you can compare what I did with what you did.

I went back to the same web page & collected the HLS master manifest (m3u8) from the Network Monitor.  I ran it through ffprobe.  That log is attached.  There were 2 surprises in there.  First was that there are actually 4 variants with this video.  VDH is showing us only 3.  I believe that is because the setting in the Video qualities for minimum resolution to recognize is 240p.  This setting should offer a value of 0, which is not an available option at the moment.  I want to be able to see ALL available variants with NO lower limit.  This isn't possible the way it works now.  So there is a 4th variant here that VDH does not permit us to see.

The second surprise is that there are WebVTT captions with this.  I'm not sure how useful they would be to you, but I thought I would try to get them just to see if I could.

You will note that in the 1920x1080 variant I downloaded, there are very many corrupt packets.  An ffmpeg log is a rather lengthy file & that sort of thing is buried in the monotonous long section of the log that reports incremental progress during a download.  I don't know if your screen reader can find what I'm talking about.  I have no experience with screen readers.  I'm just glad they exist & people successfully use them, as you obviously do.  Who knows?  I might need to learn how to use one some day.  Maybe it would help to tell you that the log lines include the word "corrupt" in them.

Here's another surprise.  After downloading a certain amount of this file, ffmpeg went to sleep.  It was the same thing as what I observed doing the audio-only downloads in VDH earlier.  I didn't put a timer on it but it seemed like it slept for a longer delay than what I observed with VDH.  But, same as with VDH, after just being patient for a few minutes, ffmpeg woke back up & completed the download without error.

The resulting mp4 was the same as what I downloaded earlier with VDH.  Playing it in VLC now showed captions.  Those weren't accessible with VDH, although it is my understanding that caption support in VDH is coming, although not necessarily very soon.

I will leave it as an exercise for you, once you have a regular version of ffmpeg, to try to use it to download the lower resolution variants here.  There is the 852x480 variant that VDH let us see, plus a 426x240 variant that VDH blocks.  It might be interesting to see how ffmpeg deals with these.  My suspicion is that ffmpeg will throw out even more errors than I got with my download.

[ffprobe.txt](https://github.com/aclap-dev/video-downloadhelper/files/14162236/ffprobe.txt)
[Test mp4 Log.txt](https://github.com/aclap-dev/video-downloadhelper/files/14162241/Test.mp4.Log.txt)
[Test vtt Log.txt](https://github.com/aclap-dev/video-downloadhelper/files/14162244/Test.vtt.Log.txt)

**Reply by whocrazy** [reply]
Hi. Even when downloading these low resolution videos with the regular nightly ffmpeg, it still goes to sleep at a certain point. When it wakes up again and resumes, I'll see the error stream ends prematurely at xxxx, should be yyyy.
But, if I try pressing q to interrupt at this point while it's frozen, it says "q" command received, exiting, then it just sits there waiting to wake up.
another error I get is, AAC bitstream not in adts format and extradata missing, but, I don't understand, why does it freeze?
Yes, this is with a regular nightly build of ffmpeg.
How do I stop ffmpeg from discarding video and audio frames?
when I download the .ts files manually, the video and audio doesn't skip ahead, and firefox doesn't freeze for a few minutes like ffmpeg does. When I concatenate the segments into one big ts file after downloading using the browser method, I don't get any errors and no frames of video or audio are dropped.

**Reply by GrampaWildWilly** [reply]
I have now downloaded both the low resolution variants of this video, 426x240 & 852x480, with ffmpeg.  See the attached logs.  Both downloads are full of corrupt packets.  This has been consistent across all the variants so I have to conclude that this web site is rather sloppy about the way they encode their content.  Either that or they are using some woefully out of date software to encode their videos.  Whatever the case, the appearance of corrupt packets in these streams is entirely on the web site.  This is not an error in any of our computers, this is not an error in any of our Internet connections, this is not an error in VDH or the CoApp of any versions.  This is entirely an error on the web site.

However, despite these errors, they are not enough to deter ffmpeg from downloading this content.  I have now successfully downloaded the highest resolution variant (1920x1080) & the 2 lowest resolution variants (426x240 & 852x480) with ffmpeg without VDH.  There is only one variant left & I really don't think I need to download that one as well.  It is clear that whatever I'm doing is working.  I strongly suggest you inspect my logs.  You will see there the exact ffmpeg invocations I executed.  You need to start using the parameter set I'm using & forget about the parameter set you've been using.  Clearly, you are doing something wrong & I am doing it right.  The 2 low resolution files I just downloaded played fine in VLC.  I didn't sit & watch them all the way through.  I skipped forward at 1 minute intervals & let them play a couple of seconds at each sample point, long enough to see that I was getting both video & audio at each sample point.  All of the downloads I have done from this site have resulted in perfectly playable content.  I am not sufficiently motivated to actually watch the entirety of this in order to verify whether there are any little glitches or skips between my sample points, any visible or audible artifacts of the plethora of corrupt packets that are getting discarded.  I leave that to people who are actually interested in watching this content.

I will say again, though, that these 2 downloads just now also stopped dead after only a short spurt of data transmission.  I was doing these downloads concurrently, as you can tell from the time stamps I include in my logs.  They both completed successfully within a few seconds of each other.  In the interim, there was a gap lasting something between 3 & 4 minutes during which ffmpeg appeared as 2 active but idle tasks in Task Manager.  But then on some trigger that I certainly didn't offer, both tasks woke back up & the downloads completed successfully.  @paulrouget has said he is looking into this.  I'm beginning to think this is something weird having to do with this web site.  I regularly download things with ffmpeg from other sites -- one particular pay site every night -- & there is no delay like this.  So I think even this delay is down to the web site.  There are some other reports in this forum of delays.  I'm not sure those other folks have been waiting the 3-4 minutes that seem to be required to allow their downloads to resume.  Even 3-4 minutes, quite an unacceptably large delay as it is, might not always be enough.  It does seem rather suspicious that suddenly, a bunch of web sites would all start imposing this annoying delay in their downloads.  I can't explain it.  Perhaps if I got a newer ffmpeg, these delays would go away.  I'm sure I will eventually get a new ffmpeg.  I usually do that about every 6-8 months.  I must say that even when I do, I will not be revisiting this problem.  All I can say is that patience is in order & you will be rewarded.

[ffprobe.txt](https://github.com/aclap-dev/video-downloadhelper/files/14172714/ffprobe.txt)
[Test 426x240 mp4 Log.txt](https://github.com/aclap-dev/video-downloadhelper/files/14172719/Test.426x240.mp4.Log.txt)
[Test 852x480 mp4 Log.txt](https://github.com/aclap-dev/video-downloadhelper/files/14172720/Test.852x480.mp4.Log.txt)

**Reply by GrampaWildWilly** [reply]
@whocrazy You keep mentioning that you get the raw ts streams.  You are expending extra effort when you do that.  The raw ts data is what ffmpeg downloads in my test.  You can see this in my logs.  Each chunk that ffmpeg reports on a line that includes the word "Opening" is a .ts fragment.  Inspect the URL carefully.  You will see .ts in each one.  It also includes a sequential number so you can tell how many chunks it took to construct the full video.  You should just let ffmpeg mux these for you.  I think mux is the right verb here.  I'm not entirely sure.  Whatever the right verb is, you should let ffmpeg do that work for you.  My tests prove that ffmpeg can & will do it for you.  It turns these streams of .ts chunks into a complete mp4.  Pretty much every download I've ever done with ffmpeg gets .ts chunks.  That's the way HLS streaming works.

You seem to be stuck on your set of parameters you give to ffmpeg.  I am going to guess that you used some AI chatbot to suggest that command to you.  You should not have blind faith (here we go again with common expressions that might seem indelicate) in whatever an AI bot tells you.  They are not infallible.  Ditto for things you find from web searches.  Much discussion of ffmpeg goes back years, sometimes over 10 years.  You need to apply some personal evaluation to anything you read.  The older the content, the more you need to evaluate it.  You are using a bunch of parameters I've never even considered using.  Your downloads are failing.  My downloads are succeeding.  What is your unavoidable conclusion?

**Reply by whocrazy** [reply]
I do not use AI chat bots for anything.  It was an extension in firefox called the stream detector.
It scans the URL and parameters to the m3u8 stream, then it will let you copy the URL and the parameters to the clipboard.
It's only certain videos that VDH and ffmpeg hiccup on, that's why you are getting different results to me, because you might be downloading different videos than I am.

**Reply by GrampaWildWilly** [reply]
I see.  Well, it seems Stream Detector may be getting some things wrong.  And the downloads I have done here have been from the viewer discretion advised URL you posted.  If you've been downloading other things, that's new information.  I can try other things so we can compare results.  But don't give me output from Stream Detector or anything else.  Just give me URLs of web pages on which videos are hosted.

**Comment by jcfrog** [comment]
I had 3 errors on the 3 resolutions for this one https://live.childrenshealthdefense.org/chd-tv/events/the-chd-bus-tour/flu-shot-injured-me/

**Reply by GrampaWildWilly** [reply]
OK Jérôme @jcfrog I'm not ignoring you.  I went to this page & attempted to download all 3 variants VDH offered.  I got the same error you did.  At least, I assume it was the same error.  So I did the usual thing & found the HLS master manifest for this thing in the Network Monitor.  I ran that through ffprobe (log attached).  As with @whocrazy's video, there are WebVTT captions with this AND a FOURTH variant that VDH refuses to display because the "Video qualities" setting is causing problems (documented here: https://github.com/aclap-dev/video-downloadhelper/discussions/41).  I downloaded all 4 variants plus the captions using ffmpeg (logs attached).  Since the download times of these 5 downloads ranged from 4 seconds to 18 seconds, none of them went to sleep.  They all completed before whatever is causing the sleep delays in the earlier downloads could interfere here.  All 4 mp4 files play without a problem, all 4 with the captions I downloaded.  I did not sit & watch this.  I just skimmed it, sampling it at one-minute intervals.  I had video, audio, & captions at each sample point.  It is possible there are glitches between my sample points.  I am not sufficiently motivated to actually watch this all the way through to determine that for certain.  I leave that as an exercise for somebody who is actually interested in this content.

So here's an example in which VDH refuses to download this but ffmpeg has no problems with it.  I will put forward a guess, and I have proposed this before, that VDH needs to supply this parameter on its ffmpeg invocations:

-fflags +discardcorrupt

You need to discard corrupt packets.  A corrupt packet is not a good enough reason to abort a download.  Just discard it.  Users aren't interested in such an error.  Just throw the packet away.  That's why ffmpeg provides this switch.  My experiment here proves that it harms nothing to toss the corrupt packets.  File corruption seems to be the norm on this web site.  Every ffmpeg download I've done from here, no matter if VDH succeeds or fails to get the same video, has contained corrupt packets.  I am less than impressed with the way these people run their web site.  At least their download speeds are decent.  But VDH needs to be discarding corrupt packets.  This should not be a setting that a user can set.  Just do it.

So there's 2 problems in VDH that need to be fixed:
- Allow 0 in that setting so all variants can be displayed, regardless of resolution, should the user choose that.
- Add -fflags +discardcorrupt to your invocation of ffmpeg so you discard corrupt packets.

[ffprobe.txt](https://github.com/aclap-dev/video-downloadhelper/files/14174116/ffprobe.txt)
[Captions (ffmpeg) vtt Log.txt](https://github.com/aclap-dev/video-downloadhelper/files/14174139/Captions.ffmpeg.vtt.Log.txt)
[426x240 (ffmpeg) mp4 Log.txt](https://github.com/aclap-dev/video-downloadhelper/files/14174124/426x240.ffmpeg.mp4.Log.txt)
[640x360 (ffmpeg) mp4 Log.txt](https://github.com/aclap-dev/video-downloadhelper/files/14174126/640x360.ffmpeg.mp4.Log.txt)
[852x480 (ffmpeg) mp4 Log.txt](https://github.com/aclap-dev/video-downloadhelper/files/14174131/852x480.ffmpeg.mp4.Log.txt)
[1280x720 (ffmpeg) mp4 Log.txt](https://github.com/aclap-dev/video-downloadhelper/files/14174132/1280x720.ffmpeg.mp4.Log.txt)

**Reply by whocrazy** [reply]
the only trouble with throwing away those corrupt packets is that bits of audio and video get chopped off, for example with the video "flu shot injured me", there's a problem when the lady talks about respiratory issues, the word "respiratory" gets cut out in mid speech.  I've not listened to the rest of the video, so there might be more problems where speech is shortened or cut off.
How do I tell ffmpeg to keep the corrupt packets and ignore the errors? I want a flawless video with flawless audio, if firefox can play it without hiccuping, why can't ffmpeg be made to do the same?

**Reply by GrampaWildWilly** [reply]
I don't believe ffmpeg can be made to retain corrupt packets.  If you don't tell ffmpeg to discard them, it quits on the first one it hits.  If you tell it to discard them, then it goes all the way to the end.  But corrupt packets aren't something you want.  If the content has corruption in it, you can't correct that.  Only the owner of the content can do that.  Everything on this site seems to contain corrupt packets.  If the words being spoken are still intelligible, that is probably the best you are going to get.  Why does it play fine in the browser window?  I don't know.  If you want to download this stuff, you're going to have to deal with corrupt packets one way or another.

**Reply by GrampaWildWilly** [reply]
You do know that VDH can download just the audio track of something now, don't you?  I mentioned it above.  Is that not something that would interest you?

**Reply by whocrazy** [reply]
How do I download just the audio track?
I went to the conversion output section and chose the mp3 option, but when I bring up the menu, it doesn't say anything about downloading audio only.

**Reply by whocrazy** [reply]
oh dear.
There was an item I had to click on that my screen reader wasn't acknowledging it was there.
I then clicked on the option download audio only in mp3, and the file successfully downloaded and I could play it, but, like with the video, it too had issues with audio cutting off mid speech, when she talks about her respiratory issues, the "respiratory" gets partially spoken and then cuts out and goes to the next part of the sentence.  Looks like I'll have to download the video the messy way if I don't want some words in the audio being interrupted, other videos with these issues will have to be obtained in the same fashion with down them all, then a merge session with ffmpeg, and then played on VLC or GOM.

**Reply by GrampaWildWilly** [reply]
OK.  You found the function for audio only.  Good.  As for the glitches in the audio, I can't explain that.  I guess if you get satisfactory results your "messy" way, as you put it, I guess that's what you have to do.

**Reply by GrampaWildWilly** [reply]
I'm not sure there is actually a problem at 6:30.  I listened to about 6:20-6:40 on the web site.  Then I downloaded it again with VDH & listened to the same stretch again.  She is choking back tears & pausing a few times in there.  You might think you're missing something but I'm not seeing or hearing any skips there.  Without the visual feedback of seeing her lips moving, you might think there's skips there, but it doesn't look like it to me.  Maybe I'm not looking at the right place.

Also, I'm not finding around 5 minutes where she's talking about chills.  That word doesn't show up in the captions at all.  But then, I suspect it's possible these captions were machine-generated so there might be some inaccuracies here & there.

I've gotten the captions again as well so if there's a certain word or a couple of words where it skips, I can find it in the captions & watch that particular part of the video to see what's going on.  With them being, as you said, so emotional, the speech is not smooth & continuous.  You might be thinking there's skips where they're just pausing to take a deep breath or wipe away tears before continuing.  Maybe, maybe not.  I'd really like to get to the bottom of this.

**Reply by GrampaWildWilly** [reply]
Oh wait.  The chill thing is in the other video.  Yes, that one skips rather badly around 5:00 & 6:30 in the video downloaded by ffmpeg.  I don't have any easy answer for that one.

**Reply by whocrazy** [reply]
if I download the video with ffmpeg, the one about the hospital killing skips badly about 0:30, when she mentions the emergency trials, so it looks like either, I'll have to download the segments manually and then merge them, or run the m3u8 through vlc media player and listen to it that way.  The videos are hosted on a platform called boxcast.
This is the only video platform I've come across with this corruption issue, all the other video platforms I've encountered, so far, ffmpeg and VDH can get them without a problem.

**Comment by paulrouget** [comment]
There's a lot of text here. Can someone summarize what's going on?

**Reply by GrampaWildWilly** [reply]
ffmpeg is retrieving content that VDH is failing on.  Also,  ffprobe on HLS master manifests is showing variants that VDH is not.  Also we're seeing stalls, but you have another discussion dealing with that.  Also VDH is not discarding corrupt packets.  But there's no substitute for reading what's posted.
